WeatherForcasterApp/public/js/app/views/desktopViews/AppsEntryPage.js

```diff
@@ -37,7 +37,10 @@ define([
             }
         },
         showCityData: function () {
-            var city_names = $(".tag span").text().split(/\s+/)
+            var city_names = [];
+            $(".tag span").each(function(){
+                city_names.push($(this).text());
+            });

```

Comment:
this code snippet will append each span element's text into array of city names.